src/ops/arithmetic.rs

```diff
@@ -0,0 +1,875 @@
+use crate::array::Array;
+use crate::stream::StreamOrDevice;
+use mlx_macros::default_device;
+
+impl Array {
+    /// Element-wise absolute value.
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    /// use mlx::Array;
+    /// let array = Array::from_slice(&[1i32, 2, -3, -4, -5], &[5]);
+    /// let mut result = array.abs();
+    ///
+    /// result.eval();
+    /// let data: &[i32] = result.as_slice();
+    /// // data == [1, 2, 3, 4, 5]
+    /// ```
+    ///
+    /// # Params
+    ///
+    /// - stream: stream or device to evaluate on
+    #[default_device]
+    pub fn abs_device(&self, stream: StreamOrDevice) -> Array {
+        let ctx = stream.as_ptr();
+
+        unsafe { Array::from_ptr(mlx_sys::mlx_abs(self.c_array, ctx)) }
+    }
+
+    /// Element-wise addition.
+    ///
+    /// Add two arrays with <doc:broadcasting>.
+    ///
+    /// # Example
+    /// ```rust
+    /// use mlx::Array;
+    /// let a = Array::from_slice(&[1.0, 2.0, 3.0], &[3]);
+    /// let b = Array::from_slice(&[4.0, 5.0, 6.0], &[3]);
+    /// let mut c = a.add_device(&b, Default::default());
+    ///
+    /// c.eval();
+    /// let c_data: &[f32] = c.as_slice();
+    /// // c_data == [5.0, 7.0, 9.0]
+    /// ```
+    ///
+    /// # Params
+    ///
+    /// - other: array to add
+    /// - stream: stream or device to evaluate on
+    pub fn add_device(&self, other: &Array, stream: StreamOrDevice) -> Array {
```

Comment:
What if the shapes of the two arrays are not broadcastable? We probably also want to check combination of two `dtype`s for binary ops that are supported

Reply:
mlx doesn't have such check, it promotes types and checks for broadcast